blobstore-fs/src/fs_utils.rs

```diff
@@ -3,7 +3,10 @@ use std::vec::Vec;
 
 /// Traverses a file system starting at location `root` and returning a list of all directories
 /// contained in that directory, recursively, relative to the original root at level 0.
-pub fn all_dirs(root: &Path, prefix: &Path) -> Vec<PathBuf> {
+pub fn all_dirs(root: &Path, prefix: &Path, depth: u32) -> Vec<PathBuf> {
+    if depth > 1000 {
+        return vec![];
```

Comment:
Returning an empty list is a little confusing. Should this function return a `Result<Vec<PathBuf>>` instead?

Also, while you're here.... We shouldn't be panicking below, when failing to read a directory

Reply:
This is recursive so what it'll do is return an empty list for directories when the depth is over 1000, but we're actually accumulating the list below

Reply:
_realistically_ anyways this is only hit when someone lists a directory with one _thousand_ nested folders beneath it and they have a need for it

Reply:
Ah, thank you 👍